process/factory/shard/preProcessorsContainerFactory.go

```diff
@@ -0,0 +1,159 @@
+	store               dataRetriever.StorageService
+	marshalizer         marshal.Marshalizer
+	hasher              hashing.Hasher
+	keyGen              crypto.KeyGenerator
+	singleSigner        crypto.SingleSigner
+	multiSigner         crypto.MultiSigner
+	dataPool            dataRetriever.PoolsHolder
+	addrConverter       state.AddressConverter
+	chronologyValidator process.ChronologyValidator
+	txProcessor         process.TransactionProcessor
+	scProcessor         process.SmartContractProcessor
+	scResultProcessor   process.SmartContractResultProcessor
+	interProcessor      process.IntermediateTransactionHandler
+	accounts            state.AccountsAdapter
+	requestHandler      process.RequestHandler
+}
+
+// NewPreProcessorsContainerFactory is responsible for creating a new preProcessors factory object
+func NewPreProcessorsContainerFactory(
+	shardCoordinator sharding.Coordinator,
+	store dataRetriever.StorageService,
+	marshalizer marshal.Marshalizer,
+	hasher hashing.Hasher,
+	dataPool dataRetriever.PoolsHolder,
+	addrConverter state.AddressConverter,
+	accounts state.AccountsAdapter,
+	requestHandler process.RequestHandler,
+	txProcessor process.TransactionProcessor,
+	scProcessor process.SmartContractProcessor,
+	scResultProcessor process.SmartContractResultProcessor,
+	interProcessor process.IntermediateTransactionHandler,
+) (*preProcessorsContainerFactory, error) {
+
+	if shardCoordinator == nil {
+		return nil, process.ErrNilShardCoordinator
+	}
+	if store == nil {
+		return nil, process.ErrNilStore
+	}
+	if marshalizer == nil {
+		return nil, process.ErrNilMarshalizer
+	}
+	if hasher == nil {
+		return nil, process.ErrNilHasher
+	}
+	if dataPool == nil {
+		return nil, process.ErrNilDataPoolHolder
+	}
+	if addrConverter == nil {
+		return nil, process.ErrNilAddressConverter
+	}
+	if txProcessor == nil {
+		return nil, process.ErrNilTxProcessor
+	}
+	if accounts == nil {
+		return nil, process.ErrNilAccountsAdapter
+	}
+	if scProcessor == nil {
+		return nil, process.ErrNilSmartContractProcessor
+	}
+	if scResultProcessor == nil {
+		return nil, process.ErrNilSmartContractResultProcessor
+	}
+	if interProcessor == nil {
+		return nil, process.ErrNilIntermediateTransactionHandler
+	}
+	if requestHandler == nil {
+		return nil, process.ErrNilRequestHandler
+	}
+
+	return &preProcessorsContainerFactory{
+		shardCoordinator:  shardCoordinator,
+		store:             store,
+		marshalizer:       marshalizer,
+		hasher:            hasher,
+		dataPool:          dataPool,
+		addrConverter:     addrConverter,
+		txProcessor:       txProcessor,
+		accounts:          accounts,
+		scProcessor:       scProcessor,
+		scResultProcessor: scResultProcessor,
+		interProcessor:    interProcessor,
+		requestHandler:    requestHandler,
+	}, nil
+}
+
+func (ppcm *preProcessorsContainerFactory) Create() (process.PreProcessorsContainer, error) {
+	container := containers.NewPreProcessorsContainer()
+
+	key, preproc, err := ppcm.createTxPreProcessor()
```

Comment:
Why is key returned from this function? It is not actually used there, so it's not useful. You can just simply return two values (preproc and err) and do `err = container.Add(block.TxBlock, preproc)`

Reply:
done